design-proposals/klip-8-interactive-queries.md

```diff
@@ -131,13 +131,129 @@ Additionally:
 * Key movement should be handled transparently: if a query launches and a targeted key physically moves, KSQL should redirect the query to the appropriate KSQL server to obtain the correct result.
 * For interactive queries against windowed queries, the user may specify a time range via the `WINDOWSTART` and `WINDOWEND` psuedocolumns. For example, `SELECT * FROM windowed_aggregate WHERE ROWKEY = 'rowkey' AND WINDOWSTART >= start_ts AND WINDOWEND < end_ts`
 
-## Open questions
+---
+
+# Syntax changes
+
+All KSQL query syntax currently yields streaming results because KSQL only supports streaming queries. As a result, adding support for point-in-time queries necessitates a means to unambiguously differentiate between streaming and finite query results. The most straightforward approach to this is to use syntax to indicate streaming versus PIT queries. And since there are only two fundamental query forms, it is only necessary to add KSQL syntax to represent one of them.
+
+We therefore propose making minimal syntax changes to represent *streaming* queries. The principal reason for using new syntax for streaming queries is that PIT queries can then use syntax that is as close to the SQL standard as possible, making KSQL interoperable with tooling that integrates with SQL-based systems (i.e. dashboarding frontends).
+
+After introducing any syntax changes, KSQL's syntax must make sense in each of the following contexts:
+
+* Streaming query on a stream
+* Streaming query on a table
+* Point-in-time query on a stream
+* Point-in-time query on a table
+* CTAS/CSAS defined by any of the above
+
+*Note that not all of these query forms need to be supported (initially, or ever), but a sensible design should allow for all of them with minimal ambiguity.*
+
+The proposed syntax changes will now be specifically described.
+
+## Proposed syntax changes
+
+Streaming queries may be identified via the `EMIT CHANGES` query modifier:
+
+```sql
+SELECT ... FROM stream EMIT CHANGES;
+```
+
+Inspired by `EMIT STREAM` from this excellent [SIGMOD paper](https://arxiv.org/pdf/1905.12133.pdf), as well as @big-andy-coates' [brainstorm](https://github.com/confluentinc/ksql/pull/3117#issuecomment-520524284), `EMIT CHANGES` has been proposed in favor of `EMIT STREAM` for the following reasons:
```

Comment:
+1 I, too, favor not using the term `STREAM` for EMIT for the reasons specified.